### PR TITLE
Make every configuration optional (fixes #42)

### LINKED ISSUE #42: Refactor settings with defaults
* [x] Centralize default values for all settings
* [x] Provide way to easy add custom default values
* [x] Override defaults with environment variables ``CLIQUET_STORAGE_URL``

For example : 

```python
from cliquet import initialize_cliquet, DEFAULT_CONFIG

app_defaults = DEFAULT_CONFIG.copy()
app_defaults['captain.age']  = 42

initialize_cliquet(config, __version__, defaults=app_defaults)
```

Use ``registry.setting('setting_name')``

```python
def read_setting(settings, defaults):
    def with_default(name):
         env_name = name.replace('.', '_').upper()
         setted = settings.get(name, defaults.get(name, ''))
         value = os.getenv(env_name, setted)
         if not value:
                  warnings.warn('Value not set for %s' % name)
    return lamdba name: with_default(name)

registry.setting = read_setting(settings)
```

### PR DESCRIPTION
**Pull request by leplatrem**
fixes #42 

While wrinting the documentation, I realize that configuration could have default everywhere

### CODE REVIEW

File: cliquet/__init__.py [hunk 1]
```diff
@@ -154,14 +156,16 @@ def includeme(config):
...
     config.registry.storage = storage.load_from_config(config)
 
-    session = config.maybe_dotted(settings['cliquet.session_backend'])
+    session = config.maybe_dotted(settings.get('cliquet.session_backend',
+                                               DEFAULT_SESSION_BACKEND))
     config.registry.session = session.load_from_config(config)
 
-    config.registry.project_name = settings['cliquet.project_name']
-    config.registry.project_docs = settings['cliquet.project_docs']
+    config.registry.project_name = settings.get('cliquet.project_name', '')
```
Comment (Natim) [inline cliquet/__init__.py]: Not sure for this one.

File: cliquet/__init__.py [hunk 2]
```diff
@@ -19,6 +19,8 @@
 from cliquet.utils import msec_time
 
 DEFAULT_OAUTH_CACHE_SECONDS = 5 * 60
+DEFAULT_STORAGE_BACKEND = 'cliquet.storage.postgresql'
```
Comment (Natim) [inline cliquet/__init__.py]: Not sure if the default should be postgresql

**Comment by Natim** [comment]
Refs https://github.com/mozilla-services/readinglist/issues/81

File: cliquet/__init__.py (hunk 1 above)
Comment (Natim) [inline cliquet/__init__.py]: I mean the project_name configuration should be mandatory and not default to empty string.

File: cliquet/__init__.py (hunk 2 above)
Comment (leplatrem) [inline cliquet/__init__.py]: ah yes, and that would mean it should  not be in extra_require then !

File: cliquet/__init__.py (hunk 1 above)
Comment (leplatrem) [inline cliquet/__init__.py]: Why ? it is used in hello view only !

Default to cliquet ?

Comment (ametaireau) [inline cliquet/__init__.py]: Aim is to fail if it's not defined so that cliquet users define it. We can also fallback to a default value but in this case issuing a warning would help.

Comment (Natim) [inline cliquet/__init__.py]: Maybe we should do the same thing as we do with version for the default value, use the default package name.

Comment (ametaireau) [inline cliquet/__init__.py]: We don't do that anymore in cliquet: cliquet doesn't make any assumptions about the version, and the client must provide it explicitely.

Comment (Natim) [inline cliquet/__init__.py]: I mean passing it as we do for version rather than reading it from the conf

Comment (ametaireau) [inline cliquet/__init__.py]: Ah okay, I understand your proposal I think. Let me rephrase so I'm sure we're on the same page here.

Since we're already passing the version information to cliquet in `initialize_cliquet`, we should reuse this version to display it on the home page.

If that's it, I concur :)

Comment (Natim) [inline cliquet/__init__.py]: > we should reuse this version to display it on the home page.
Isn't it already the case?

Comment (ametaireau) [inline cliquet/__init__.py]: Hmm, I'm almost all wrong. We're already doing that (I wrote the patch for it, meh).

Your proposal is to do the same thing for the name. If yes, then I believe we should use that for default, but we should let the possibility to the user to configure another name (for instance,  "cloud storage" is a project name and "kinto" is the codename. They aren't the same).

Comment (Natim) [inline cliquet/__init__.py]: > Your proposal is to do the same thing for the name. If yes, then I believe we should use that for default

Totally agree :+1:

**Comment by leplatrem** [comment]
r?

**Comment by leplatrem** [comment]
If you agree, I update this PR to include #42 and implement what you suggest about project name

**Comment by Natim** [comment]
Yes ok.

**Comment by Natim** [comment]
A warning is not exactly what I suggested about project name :)

**Comment by leplatrem** [comment]
Yes I know

**Comment by leplatrem** [comment]
@ametaireau @Natim r?

Here is what I did: 
* Read settings from environment
* Centralize defaults
* Cleaned tests settings to use defaults
* Got rid of ``settings.get`` since I assume we have defaults for everything
* Got rid of registry usage for hello view value (*thought it has better consistency*)
* Added project_name to initialize_cliquet (*not a fan of this btw, I would have liked to keep this method very very stupid*)
* Warn if no project name
* Raise error is no project version (can also be specified in settings)
